### PR TITLE
Aligned doc output format with repro scripts

### PR DESCRIPTION
**Pull request by wu-ming233**
Aligned regression documentation output format with the repro scripts. The unified format is:
```
run.collection-name.condition-name.topic-name.txt
```
where `collection-name` in {"msmarco-v2.1-doc", "msmarco-v1-passage", "beir"}, `condition-name` is specified in the yaml files under `src/main/resources/reproduce`, and `topic-name` is specified by the corpus.

Additionally, the repro scripts now run the same amount of cases as in the documentation.

### CODE REVIEW

File: docs/fatjar-regressions/fatjar-regressions-v0.36.1-SNAPSHOT.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ Here are the instructions for reproducing runs on the MS MARCO V2.1 segmented do
 ```bash
 TOPICS=(msmarco-v2-doc-dev msmarco-v2-doc-dev2 trec2021-dl trec2022-dl trec2023-dl rag24-raggy-dev); for t in "${TOPICS[@]}"
 do
-    java -cp $ANSERINI_JAR io.anserini.search.SearchCollection -index msmarco-v2.1-doc-segmented -topics $t -output $OUTPUT_DIR/run.msmarco-v2.1-doc-segmented.bm25.${t}.txt -threads 16 -bm25 -hits 10000 -selectMaxPassage -selectMaxPassage.delimiter "#" -selectMaxPassage.hits 1000
```
Comment (lintool) [inline docs/fatjar-regressions/fatjar-regressions-v0.36.1-SNAPSHOT.md:105]: actually, `doc-segmented` is correct. The two MS MARCO 2.1 conditions are `doc` and `doc-segmented`.

File: docs/fatjar-regressions/fatjar-regressions-v0.36.1-SNAPSHOT.md [hunk 2]
```diff
@@ -383,17 +383,17 @@ The following snippet will generate the complete set of results for BEIR:
 CORPORA=(trec-covid bioasq nfcorpus nq hotpotqa fiqa signal1m trec-news robust04 arguana webis-touche2020 cqadupstack-android cqadupstack-english cqadupstack-gaming cqadupstack-gis cqadupstack-mathematica cqadupstack-physics cqadupstack-programmers cqadupstack-stats cqadupstack-tex cqadupstack-unix cqadupstack-webmasters cqadupstack-wordpress quora dbpedia-entity scidocs fever climate-fever scifact); for c in "${CORPORA[@]}"
 do
     # "flat" indexes
-    java -cp $ANSERINI_JAR io.anserini.search.SearchCollection -index beir-v1.0.0-${c}.flat -topics beir-${c} -output $OUTPUT_DIR/run.beir.${c}.flat.txt -bm25 -removeQuery
```
Comment (lintool) [inline docs/fatjar-regressions/fatjar-regressions-v0.36.1-SNAPSHOT.md:386]: Actually, `beir.${c}` is more accurate. BEIR has sub-collections, and we want to keep BEIR and the subcollections together.

File: src/main/resources/reproduce/msmarco-v1-passage.yaml [hunk 3]
```diff
@@ -9,19 +9,14 @@ conditions:
         eval_key: msmarco-passage.dev-subset
         scores:
           - MRR@10: 0.1840
-            R@1K: 0.8526
```
Comment (lintool) [inline src/main/resources/reproduce/msmarco-v1-passage.yaml:12]: let's keep existing metrics please.

**Review by lintool — CHANGES_REQUESTED** [review]

File: docs/fatjar-regressions/fatjar-regressions-v0.36.1-SNAPSHOT.md (hunk 1 above)
Comment (wu-ming233) [reply docs/fatjar-regressions/fatjar-regressions-v0.36.1-SNAPSHOT.md:105]: For the current setup, the output file names are of the form `run.collection-name.condition-name.topic-name.txt`. I would modify it such that `collection-name` is `msmarco-v2.1`, `condition-name` in {`doc`, `doc-segmented`}, so the resulting file names would look like, for example:
`run.msmarco-v2.1.doc.trec2021-dl.txt`
`run.msmarco-v2.1.doc-segmented.trec2021-dl.txt`
Would this be desired?

Comment (lintool) [reply docs/fatjar-regressions/fatjar-regressions-v0.36.1-SNAPSHOT.md:105]: yup!

File: src/main/java/io/anserini/reproduce/RunMsMarco.java [hunk 4]
```diff
@@ -86,20 +86,15 @@ public MsMarcoMetricDefinitions() {
       // msmarco-v1-passage definitions
       Map<String, String> msmarcoDevSubsetMetrics = new HashMap<>();
       msmarcoDevSubsetMetrics.put("MRR@10", "-c -M 10 -m recip_rank");
-      msmarcoDevSubsetMetrics.put("R@1K", "-c -m recall.1000");
```
Comment (lintool) [inline src/main/java/io/anserini/reproduce/RunMsMarco.java:89]: Please keep all of these below

**Review by lintool — CHANGES_REQUESTED** [review]

File: src/main/java/io/anserini/reproduce/RunMsMarco.java (hunk 4 above)
Comment (wu-ming233) [reply src/main/java/io/anserini/reproduce/RunMsMarco.java]: Got it. Sorry about that

File: docs/reproduce/msmarco-v1-passage.html [hunk 5]
```diff
@@ -218,18 +218,18 @@ <h1 class="mb-3">MS MARCO V1 Passage</h1>
...
-<pre><code>tools/eval/trec_eval.9.0.4/trec_eval -c -l 2 -m map dl19-passage \
-  runs/run.msmarco.bm25-default.dl19.txt
+<pre><code>tools/eval/trec_eval.9.0.4/trec_eval null dl19-passage \
+  runs/run.msmarco.bm25.dl19.txt
 tools/eval/trec_eval.9.0.4/trec_eval -c -m ndcg_cut.10 dl19-passage \
-  runs/run.msmarco.bm25-default.dl19.txt
-tools/eval/trec_eval.9.0.4/trec_eval -c -l 2 -m recall.1000 dl19-passage \
-  runs/run.msmarco.bm25-default.dl19.txt
+  runs/run.msmarco.bm25.dl19.txt
+tools/eval/trec_eval.9.0.4/trec_eval null dl19-passage \
```
Comment (lintool) [inline docs/reproduce/msmarco-v1-passage.html:231]: why null here?

File: src/main/java/io/anserini/reproduce/RunMsMarco.java [hunk 6]
```diff
@@ -66,14 +66,14 @@ public static void main(String[] args) throws Exception {
     }
     switch (MsMarcoArgs.MsMarcoVersion) {
         case 2:
-            COLLECTION = "msmarco-v2.1-doc";
+            COLLECTION = "msmarco-v2.1";
```
Comment (lintool) [inline src/main/java/io/anserini/reproduce/RunMsMarco.java:69]: can we be consistent here and change the parameter to either "msmarco-v1-passage" or "msmarco-v2.1" as opposed to "1" and "2"? We'll add others later...

File: src/main/java/io/anserini/reproduce/RunRepro.java [hunk 7]
```diff
@@ -64,8 +66,15 @@ public void run() throws StreamReadException, DatabindException, IOException, In
...
         System.out.println("  - topic_key: " + topic.topic_key + "\n");
 
-        final String output = String.format("runs/run.%s.%s.%s.txt", COLLECTION, condition.name,
-            topic.topic_key);
+        final String output;
+        if (sortByCondition) {
+          output = String.format("runs/run.%s.%s.%s.txt", COLLECTION, condition.name,
+          topic.topic_key);
+        }
+        else {
```
Comment (lintool) [inline src/main/java/io/anserini/reproduce/RunRepro.java:74]: house style:
```
} else {
```

**Review by lintool — CHANGES_REQUESTED** [review]

File: docs/reproduce/msmarco-v1-passage.html (hunk 5 above)
Comment (wu-ming233) [reply docs/reproduce/msmarco-v1-passage.html]: Not sure what happened here, but rebuilding solved it.

**Review by lintool — APPROVED** [review]

**Pull request merged**